src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -84,16 +83,19 @@ private async Task<string> GetRegistrationEndpointAsync()
         }
 
         /// <summary>
-        ///     Download one NuGet package and extract it to the target directory.
+        /// Download one NuGet package to the target directory and (optionally) extract it.
         /// </summary>
-        /// <param name="purl"> Package URL of the package to download. </param>
-        /// <returns> n/a </returns>
+        /// <remarks>The target directory is defined when creating the <see cref="NuGetProjectManager"/> in the subdirectory named `nuget-{packagename}@{packageversion}`</remarks>
+        /// <param name="purl">The <see cref="PackageURL"/> of the package to download.</param>
+        /// <param name="doExtract">If the contents of the .nupkg should be extracted into a directory.</param>
+        /// <param name="cached">If the downloaded contents should be retrieved from the cache if they exist there.</param>
+        /// <returns>An <see cref="IEnumerable{String}"/> list of the path(s) the contents were downloaded to.</returns>
         public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl, bool doExtract, bool cached = false)
         {
-            Logger.Trace("DownloadVersion {0}", purl?.ToString());
+            Logger.Trace("DownloadVersion {0}", purl.ToString());
 
-            string? packageName = purl?.Name;
-            string? packageVersion = purl?.Version;
+            string? packageName = purl.Name;
```

Comment:
Is the PackageUrl no longer nullable here?

Reply:
It isn't. You need a PackageUrl with a name and version specified in order to download said package version.

Reply:
I think the point was that the purl itself is coming from a public api - even though you have declared it to not be null in the interface you can't trust that a user won't give you a null.

To be fair though it would actually crash on line 95 when you print the name wihout the null guard.